### DISCUSSION TITLE
How to load local files in compose multiplatform project?

### DISCUSSION THREAD

**Discussion by findjigar** [coil-kt/coil#2039] [General]
We've local images in Android's `drawable-nodpi` folder and ios app's `Assets.xcassets` folder.

Now we have defined name of the images in common code of compose multiplatfrom project using `String` name of the image.

For example, for image named `about_app_icon`, how do we load this image via common code using `AsyncImage` API? We get the name of the image via ViewModel in composable like `viewModel.icon`

PS: Is there a better way to load local images?

**Comment by colinrtwhite** [comment]
Coil is able to load anything on the file system through Okio or any network URLs via Ktor from common code. Assets aren't well supported in multiplatform at the moment unfortunately. I'm waiting to see how Compose Multiplatform's [resource support](https://github.com/JetBrains/compose-multiplatform/tree/a4cb97a6bff0ce6d767cbb6a1deb895126e01222/components/resources/library) evolves (it's very early) before integrating with it.

In the meantime you'll have to add your own `Fetcher`s that transform a `Uri` into an `ImageSource` if you want to load resources/assets.
